pkg/cdi/spec.go

```diff
@@ -262,3 +262,75 @@ func validateSpec(raw *cdi.Spec) error {
 	}
 	return nil
 }
+
+// GenerateSpecName generates a vendor+class scoped Spec file name. The
+// name can be passed to WriteSpec() to write a Spec file to the file
+// system.
+//
+// vendor and class should match the vendor and class of the CDI Spec.
+// The file name is generated without a ".json" or ".yaml" extension.
+// The caller can append the desired extension to choose a particular
+// encoding. Otherwise WriteSpec() will use its default encoding.
+//
+// This function always returns the same name for the same vendor/class
+// combination. Therefore it cannot be used as such to generate multiple
+// Spec file names for a single vendor and class.
+func GenerateSpecName(vendor, class string) string {
+	return vendor + "-" + class
+}
+
+// GenerateTransientSpecName generates a vendor+class scoped transient
+// Spec file name. The name can be passed to WriteSpec() to write a Spec
+// file to the file system.
+//
+// Transient Specs are those whose lifecycle is tied to that of some
+// external entity, for instance a container. vendor and class should
+// match the vendor and class of the CDI Spec. driver can be left empty
+// if only a single CDI user is generating transient Spec files for the
+// given vendor/class combination on the host. Otherwise driver should
+// uniquely identify the caller among those that use the same vendor
+// and class. transientID should uniquely identify the external entity
+// within the scope of the caller/driver.
+//
+// The file name is generated without a ".json" or ".yaml" extension.
+// The caller can append the desired extension to choose a particular
+// encoding. Otherwise WriteSpec() will use its default encoding.
+func GenerateTransientSpecName(vendor, class, driver, transientID string) string {
```

Comment:
What if we remove driver from here? Caller can include it into the transientID if needed. Otherwise we'd need to explain what is it, which just seems not needed for me. Any software can call this API, not just drivers or plugins.

Reply:
I'm fine with that if we are sure that this is not going to cause any problems (in the case when multiple drivers/plugins generate Spec files with the same vendor/class combination).